protocols/imap/src/main/java/org/apache/james/imap/processor/base/UidMsnConverter.java

```diff
@@ -35,8 +36,16 @@
 
     @VisibleForTesting final ArrayList<MessageUid> uids;
 
-    public UidMsnConverter(Iterator<MessageUid> iterator) {
-        uids = Lists.newArrayList(iterator);
+    public UidMsnConverter() {
+        this.uids = Lists.newArrayList();
+    }
+
+    public synchronized void addAll(List<MessageUid> iterator) {
+        LinkedHashSet<MessageUid> messageUids = new LinkedHashSet<MessageUid>();
```

Comment:
Because it can not be sorted by calls to Collections.sort (which requires a list)

Hence the copy logics. Sorry, I did not find better.